pkgs/applications/misc/gnuradio/default.nix

```diff
@@ -45,21 +56,50 @@ stdenv.mkDerivation rec {
     export NIX_CFLAGS_COMPILE="$NIX_CFLAGS_COMPILE -Wno-unused-variable"
   '';
 
+  # Framework path needed for qwt6_qt4 but not qwt5
+  cmakeFlags =                             
+    stdenv.lib.optionals stdenv.isDarwin [ "-DCMAKE_FRAMEWORK_PATH=${qwt}/lib" ];
+
+  postBuild = 
+    stdenv.lib.optionalString stdenv.isDarwin ''
+      echo "Attempting to patch qwt-dependent libs"
+      find . "(" -name "_qtgui_swig.so" -o -name "libgnuradio-qtgui*.dylib" ")" -print -exec install_name_tool -change qwt.framework/Versions/6/qwt ${qwt}/lib/qwt.framework/Versions/6/qwt "{}" ";"
+    '';
+
   # - Ensure we get an interactive backend for matplotlib. If not the gr_plot_*
   #   programs will not display anything. Yes, $MATPLOTLIBRC must point to the
   #   *dirname* where matplotlibrc is located, not the file itself.
   # - GNU Radio core is C++ but the user interface (GUI and API) is Python, so
   #   we must wrap the stuff in bin/.
+  #
+  # May want to use makeWrapper instead of wrapProgram
+  # Also see https://github.com/NixOS/nixpkgs/issues/22688 regarding use of --prefix / python.withPackages
   postInstall = ''
     printf "backend : Qt4Agg\n" > "$out/share/gnuradio/matplotlibrc"
 
     for file in "$out"/bin/* "$out"/share/gnuradio/examples/*/*.py; do
         wrapProgram "$file" \
             --prefix PYTHONPATH : $PYTHONPATH:$(toPythonPath "$out") \
-            --set MATPLOTLIBRC "$out/share/gnuradio"
+            --set MATPLOTLIBRC "$out/share/gnuradio" ${stdenv.lib.optionalString stdenv.isDarwin " --set DYLD_FRAMEWORK_PATH /System/Library/Frameworks"}
```

Comment:
Can you add a comment pointing to https://github.com/NixOS/nixpkgs/issues/24693, when that's fixed we should be able to get rid of the `DYLD_FRAMEWORK_PATH` stuff.